### ISSUE TITLE
Per layer noise config for palettes

### ISSUE THREAD

**Issue by Astrashh** [PolyhedralDev/Terra#55]
Being able to define different noise settings per layer of blocks within a noise palette would be very handy - lets say you wanted the top layer of blocks to be organized by blobs of cellular noise, then the layer below that to be a static arrangement of blocks. This wouldn't be possible with the current config options (you could do it with absolute noise but that doesn't factor in terrain shape).

Could be configured simply like so:
```yaml
layers:
  - materials:
      - "minecraft:ice": 1
    noise:
      noise-stuff
    layers: 1
  - materials:
      - "minecraft:water": 1
    noise:
      noise-stuff
    layers: 1
```

**Comment by dfsek** [comment]
I will likely implement something similar to the example, where the noise key (which will probably be called `distribution`) is optional, and if it is not present, the layer uses the noise function defined in the root.

**Comment by dfsek** [comment]
Implemented in 20de531c8fdc3c7adb808e2baa0d0f0325cabb15
